src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -937,95 +975,104 @@ InferenceProfiler::MergePerfStatusReports(
-  summary_status.client_stats.request_count = 0;
-  summary_status.client_stats.sequence_count = 0;
-  summary_status.client_stats.delayed_request_count = 0;
-  summary_status.client_stats.duration_ns = 0;
-  summary_status.client_stats.avg_latency_ns = 0;
-  summary_status.client_stats.percentile_latency_ns.clear();
-  summary_status.client_stats.latencies.clear();
-  summary_status.client_stats.std_us = 0;
-  summary_status.client_stats.avg_request_time_ns = 0;
-  summary_status.client_stats.avg_send_time_ns = 0;
-  summary_status.client_stats.avg_receive_time_ns = 0;
-  summary_status.client_stats.infer_per_sec = 0;
-  summary_status.client_stats.sequence_per_sec = 0;
-  summary_status.client_stats.completed_count = 0;
-  summary_status.stabilizing_latency_ns = 0;
+  experiment_perf_status.client_stats.request_count = 0;
+  experiment_perf_status.client_stats.sequence_count = 0;
+  experiment_perf_status.client_stats.delayed_request_count = 0;
+  experiment_perf_status.client_stats.duration_ns = 0;
+  experiment_perf_status.client_stats.avg_latency_ns = 0;
+  experiment_perf_status.client_stats.percentile_latency_ns.clear();
+  experiment_perf_status.client_stats.latencies.clear();
+  experiment_perf_status.client_stats.std_us = 0;
+  experiment_perf_status.client_stats.avg_request_time_ns = 0;
+  experiment_perf_status.client_stats.avg_send_time_ns = 0;
+  experiment_perf_status.client_stats.avg_receive_time_ns = 0;
+  experiment_perf_status.client_stats.infer_per_sec = 0;
+  experiment_perf_status.client_stats.sequence_per_sec = 0;
+  experiment_perf_status.client_stats.completed_count = 0;
+  experiment_perf_status.stabilizing_latency_ns = 0;
 
   std::vector<ServerSideStats> server_side_stats;
   for (auto& perf_status : perf_status_reports) {
     // Aggregated Client Stats
-    summary_status.client_stats.request_count +=
+    experiment_perf_status.client_stats.request_count +=
         perf_status.client_stats.request_count;
-    summary_status.client_stats.sequence_count +=
+    experiment_perf_status.client_stats.sequence_count +=
         perf_status.client_stats.sequence_count;
-    summary_status.client_stats.delayed_request_count +=
+    experiment_perf_status.client_stats.delayed_request_count +=
         perf_status.client_stats.delayed_request_count;
-    summary_status.client_stats.duration_ns +=
+    experiment_perf_status.client_stats.duration_ns +=
         perf_status.client_stats.duration_ns;
 
     server_side_stats.push_back(perf_status.server_stats);
 
-    summary_status.client_stats.latencies.insert(
-        summary_status.client_stats.latencies.end(),
+    experiment_perf_status.client_stats.latencies.insert(
+        experiment_perf_status.client_stats.latencies.end(),
         perf_status.client_stats.latencies.begin(),
         perf_status.client_stats.latencies.end());
+    // Accumulate the overhead percentage and send rate here to remove extra
+    // traversals
```

Comment:
What does it mean to remove extra traversals?

Reply:
I have to iterate over the perf_status_reports to accumulate the overhead_pct and send_request_rate before getting the average. I just didnt want to write a new method and iterate over the reports a 2nd time(the traversal).  
Adding `over the perf_status_reports` to the comment